src/integrationTest/kotlin/com/sourcegraph/cody/util/CodyIntegrationTextFixture.kt

```diff
@@ -205,73 +202,43 @@ open class CodyIntegrationTextFixture : BasePlatformTestCase(), LensListener {
     }
   }
 
-  protected fun assertInlayIsShown() {
-    runInEdtAndWait {
-      PlatformTestUtil.dispatchAllEventsInIdeEventQueue()
-      assertTrue(
-          "Lens group inlay should be displayed", myFixture.editor.inlayModel.hasBlockElements())
-    }
+  override fun onLensesUpdate(vf: VirtualFile, codeLenses: List<ProtocolCodeLens>) {
+    synchronized(lensSubscribers) { lensSubscribers.removeAll { it(codeLenses) } }
   }
 
-  override fun onLensesUpdate(
-      lensWidgetGroup: LensWidgetGroup?,
-      codeLenses: List<ProtocolCodeLens>
-  ) {
-    synchronized(lensSubscribers) {
-      lensSubscribers.removeAll { (checkFunc, future) ->
-        if (codeLenses.find { it.command?.command == "cody.fixup.codelens.error" } != null) {
-          future.completeExceptionally(IllegalStateException("Error group shown"))
-          return@removeAll true
-        }
-
-        val hasLensAppeared = checkFunc(codeLenses)
-        if (hasLensAppeared) future.complete(lensWidgetGroup)
-        hasLensAppeared
-      }
-    }
+  fun runAndWaitForCleanState(actionIdToRun: String) {
```

Comment:
do we need `runAndWaitForCleanState`? we could use `runAndWaitForLenses` directly

Reply:
We could, but it is not obvious that `runAndWaitForLenses(actionIdToRun)` wait for *no lenses* to be visible anymore.
Or maybe I should change name of this function?